Models/ProjectModels/Admin/BookModel.php

```diff
@@ -0,0 +1,267 @@
+
+        return $data;
+    }
+
+    /**
+     * @param array $data
+     * @param string $param
+     * @throws \Exception
+     */
+    public function newItem(array $data, string $param): void
+    {
+        if ($param === 'book') {
+            $this->newBook($data, $param);
+        } else {
+            $this->newBookProperty($data, $param);
+        }
+    }
+
+    /**
+     * @param array $data
+     * @param string $param
+     * @throws \Exception
+     */
+    private function newBookProperty(array $data, string $param): void
+    {
+        $this->new($data, $param);
+    }
+
+    /**
+     * @param array $data
+     * @param string $param
+     * @throws \Exception
+     */
+    private function new(array $data, string $param): bool
+    {
+        if ($param === 'book') {
+            $conditionData['title'] = $data['title'];
+        } else {
+            $conditionData = $data;
+        }
+
+        $this->setDbMsgModel();
+        $dbResult = $this->db->select(['id'])->
+            from([$this->getTableName($param)])->condition($conditionData)->query()->fetchAll();
+        if ($dbResult) {
+            $this->msgModel->setMsg('failure', $param . '_exist', $param . '_exist');
+
+            return false;
+        }
+
+        if (!$this->db->insertData($this->getTableName($param), $data)) {
+            throw new \Exception(
+                'Failure to add new ' . $param . ', check what comes in sql string!'
+            );
+        } else {
+            $this->msgModel->setMsg('success', $param, $param);
+        }
+
+        return true;
+    }
+
+    /**
+     * @param array $data
+     * @param string $param
+     * @throws \Exception
+     */
+    private function newBook(array $data, string $param): void
+    {
+        $data['image'] = $this->createUniqueFileName('image');
+        if (!$this->moveUploadFile('image', 'book', $data['image'])) {
+            $this->msgModel->setErrorMsg('file');
+
+            return;
+        }
+
+        if (!$this->new($data, $param)) {
+            $this->deleteFile('image', 'book', $data['image']);
+        }
+    }
+
+    /**
+     * @return IDataManagement
+     * @throws \Exception
+     */
+    protected function getFileInfo(): IDataManagement
+    {
+        if (!$this->fileInfo) {
+            $this->fileInfo = DataRegistry::getInstance()->get('file');
+        }
+
+        return $this->fileInfo;
+    }
+
+    protected function getTableName(string $param): ?string
+    {
+        if (!array_key_exists($param, self::TABLE_NAMES)) {
+            throw new InvalidArgumentException(
+                'Wrong param : ' . "'$param'" . ', during choosing DB table name from const TABLE_NAMES!
+                Check this const or what comes here!'
+            );
+        }
+
+        return self::TABLE_NAMES[$param];
+    }
+
+    protected function getRequestFields(string $methodName, $tableName): ?array
+    {
+        if (array_key_exists($methodName, self::REQUEST_FIELDS)) {
```

Comment:
It would be better if initially you checked method name, throw expection of it doesn't exist and move one in a different way. 

```
protected function getRequestFields(string $methodName, string $tableName): ?array
{
    // Check if the method name exists in the REQUEST_FIELDS constant
    if (!array_key_exists($methodName, self::REQUEST_FIELDS)) {
        throw new InvalidArgumentException(
            "Invalid method name: '$methodName'."
        );
    }
    
    // Check if the table name exists for the given method
    if (!array_key_exists($tableName, self::REQUEST_FIELDS[$methodName])) {
        throw new InvalidArgumentException(
            "Invalid table name: '$tableName' for method: '$methodName'."
        );
    }

    // Return the array of fields for the given method and table
    return self::REQUEST_FIELDS[$methodName][$tableName];
}

Does it look more clear?
```